src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -76,6 +76,13 @@ public sealed override async Task ComputeRefactoringsAsync(CodeRefactoringContex
                 return;
             }
 
+            // Need to special case for expressions whose direct parent is a MemberAccessExpression since they will
+            // never introduce a parameter that makes sense in that case.
+            if (syntaxFacts.IsNameOfAnyMemberAccessExpression(expression))
```

Comment:
so, i don't have an objection to ths.   but what you're doing is building a disallow-list.  When that happens we should always ask: should this be an allow-list instead.  Specifically, should we instead be enumerating the set of cases we know are valid to extract as opposed to trying to enumerate the set of cases that should be blocking?

Reply:
I was thinking about this when looking at the code, but I feel like enumerating the cases that are allowed would be more lengthy than the blocking list.